### PR TITLE
Integrate Grom into the gem

### PR DESCRIPTION
**Pull request by aebirim**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,4 +1,9 @@
 source 'https://rubygems.org'
 
+gem 'grom', git: 'https://github.com/mattrayner/grom.git', branch: 'mattrayner/new-grom'
```
Comment (mattrayner) [inline Gemfile]: Should we release New Grom and add it as a standard dependency?

File: lib/parliament/request.rb [hunk 2]
```diff
@@ -22,7 +22,9 @@ def respond_to_missing?(method, include_private = false)
     end
 
     def get
-      Net::HTTP.get(URI(api_endpoint))
+      data = Net::HTTP.get(URI(api_endpoint))
```
Comment (mattrayner) [inline lib/parliament/request.rb]: Do we need to add error checking around this?

What happens if the API goes down? Or there's a 500 error?

File: lib/parliament/response.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
+require 'forwardable'
+
+module Parliament
+  class Response
+    include Enumerable
+    extend Forwardable
+    attr_reader :nodes
```
Comment (mattrayner) [inline lib/parliament/response.rb]: Can we add some white space around this?

File: lib/parliament/response.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+      filtered_objects = Array.new(types.size) { [] }
+
+      @nodes.each do |node|
+        type_index = types.index(node.type)
+        filtered_objects[type_index] << node unless type_index.nil?
+      end
+
+      filtered_objects.map do |objects|
+        Parliament::Response.new(objects)
+      end
```
Comment (mattrayner) [inline lib/parliament/response.rb]: Is it clear what is being returned by filter?

File: parliament.gemspec [hunk 5]
```diff
@@ -34,6 +34,7 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency 'rspec', '~> 3.0'
   spec.add_development_dependency 'rubocop'
```
Comment (mattrayner) [inline parliament.gemspec]: We should include versions for all gems we are dependent on, both for reliability and security reasons

File: spec/parliament/request_spec.rb [hunk 6]
```diff
@@ -63,9 +63,27 @@
   end
 
   describe '#get' do
```
Comment (mattrayner) [inline spec/parliament/request_spec.rb]: Should we also check our delegation? Ensuring all methods we *think* are being delegated actually are?

Comment (mattrayner) [inline spec/parliament/request_spec.rb]: Should these tests happen in the `Parliament::Response` tests?

File: lib/parliament/response.rb [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+    extend Forwardable
+    attr_reader :nodes
+    def_delegators :@nodes, :size, :each, :select, :map, :select!, :map!, :count, :length
+
+    def initialize(nodes)
+      @nodes = nodes
+    end
+
+    def filter(*types)
+      filtered_objects = Array.new(types.size) { [] }
```
Comment (mattrayner) [inline lib/parliament/response.rb]: If an empty array is passed to filter, or somehow filter objects equals `[]` should we still loop over all our nodes?

File: spec/parliament/response_spec.rb [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+  let(:nodes) { [] }
+  subject { Parliament::Response.new(nodes) }
+
+  describe '#initialize' do
+    it 'sets an instance variable for the nodes' do
+      expect(subject.instance_variable_get(:@nodes)).to eq(nodes)
+    end
+  end
+
+  describe '#filter' do
```
Comment (mattrayner) [inline spec/parliament/response_spec.rb]: Should there be a test were you filter a type that isn't present?

Comment (mattrayner) [inline spec/parliament/response_spec.rb]: Should there be a test where you call filter on an empty `Parliament::Response` object?

File: lib/parliament/request.rb (hunk 2 above)
Comment (aebirim) [inline lib/parliament/request.rb]: Please take a look we have pushed the error checking code for this

File: lib/parliament/request.rb [hunk 9]
```diff
@@ -22,7 +22,15 @@ def respond_to_missing?(method, include_private = false)
     end
 
     def get
-      Net::HTTP.get(URI(api_endpoint))
+      response = Net::HTTP.get_response(URI(api_endpoint))
+
+      if response.is_a?(Net::HTTPClientError)
```
Comment (mattrayner) [inline lib/parliament/request.rb]: these conditionals should be restructured into guard clauses such as:

```ruby
raise StandardError, 'This is a HTTPClientError' if response.is_a?(Net::HTTPClientError)
raise StandardError, 'This is a HTTPServerError' if response.is_a?(Net::HTTPServerError)

Parliament::Response.new(Grom::Reader.new(response.body).objects)
```

File: Gemfile (hunk 1 above)
Comment (katylouise) [inline Gemfile]: After discussion, we decided that until we have plugged into the front end and checked that is working as we intended, then we won't release New Grom.

File: spec/parliament/request_spec.rb (hunk 6 above)
Comment (katylouise) [inline spec/parliament/request_spec.rb]: We have added tests to check that Parliament::Response responds to the delegated methods.

File: lib/parliament/response.rb (hunk 3 above)
Comment (katylouise) [inline lib/parliament/response.rb]: Rubocop complains about whitespace here.

File: lib/parliament/response.rb (hunk 7 above)
Comment (katylouise) [inline lib/parliament/response.rb]: Added a conditional around the nodes loop which checks if types is empty.

File: lib/parliament/response.rb (hunk 4 above)
Comment (katylouise) [inline lib/parliament/response.rb]: Could we have further explanation on this, as we felt it was fairly clear what is being returned - ie. an array of Parliament::Response objects.

File: parliament.gemspec (hunk 5 above)
Comment (katylouise) [inline parliament.gemspec]: We have added the versions based on the Gemfile.lock.  Not sure if this is what you intended?

File: spec/parliament/response_spec.rb (hunk 8 above)
Comment (katylouise) [inline spec/parliament/response_spec.rb]: Added a test to check this.

Comment (katylouise) [inline spec/parliament/response_spec.rb]: Added a test to check this.

File: lib/parliament/request.rb (hunk 9 above)
Comment (katylouise) [inline lib/parliament/request.rb]: We have restructed this into guard clauses.

**Pull request merged**